### DISCUSSION TITLE
App selection when public user uploads only shows media picker and camera

### DISCUSSION THREAD

**Discussion by morkyy** [immich-app/immich#11406] [Feature Request]
### The bug

When sharing an album to public users for them to upload photos from their android phone, they get a limited app selection which only shows camera and media picker.

The option to directly open Google photos or any other photo/video gallery is not available.

You have the option to upload through media picker but it's a hassle to traverse and find pictures as it's more of a file manager rather than gallery app. There is an option through media picker to open up Google photos but that requires multiple actions from the user.

There should be the option to upload directly from Google photos or other gallery.

### The OS that Immich Server is running on

Unraid

### Version of Immich Server

Latest

### Version of Immich Mobile App

Latest

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Default as in unraid store
```

### Your .env content

```Shell
Default
```

### Reproduction steps

```bash
1.share public album link
2.open link from android phone
3.try and upload app
4.limoted app selection
...
```

**Comment by morkyy** [comment]
I'm unsure why this moved to a conversation with no comments. 
I'd imagine that the lack of any gallery app as an option when uploading pictures would be pretty major functionality required to do what Immich does?

It is not a feature request but rather a half-baked feature that its workflow is not working as intended, so a bug.

**Reply by alextran1502** [reply]
Ah sorry, I think he meant to move this to a feature request since we don't have such functionality yet to call it a bug 😅

**Reply by morkyy** [reply]
No problem! Although I believe the functionality is there. When sharing a public link to an album, public users can upload pictures to the link.

My issue is that when they try to upload from an Android device the app selection they get to pick the pictures to upload is limited to camera and media picker. This should also include gallery apps.

**Reply by mmomjian** [reply]
Yes I mean to mark it as a feature request. This would be considered a FR since it is an enhancement (adding of other library apps) to an existing feature (uploads). Issues are for things that are not working as intended. Thanks for understanding!

**Comment by bo0tzz** [comment]
I think this picker is served by the browser and Immich has no control over it

**Reply by morkyy** [reply]
> I think this picker is served by the browser and Immich has no control over it

Other image uploading services allow selection from more apps including the phone's gallery which is missing here. Also tested this across multiple browsers without change. Only on iPhones you get the gallery app option.

Unsure whether this is browser dependent but it's a bit problematic if you can't upload images from your default gallery up from all android phones.

Might be worth it to look into, to see if there's a fix on Immich side and if not submit it as a feature request on chromium or similar?
